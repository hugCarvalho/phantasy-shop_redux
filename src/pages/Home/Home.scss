.Home {
  background-color: gray;
  background-image: url("../../img/bb_02.jpg");
  background-repeat: no-repeat;
  background-attachment: scroll;
  background-size: 100%;
  color: rgba(255, 255, 255, 0.85);
  text-align: center;
  font-size: 1.9rem;
  word-break: keep-all;

  h1 {
    padding: 20px 0;
    font-size: 4rem;
  }

  .typing {
    position: relative;
    // margin-left: 10px;
    // background-color: yellowgreen;
    margin-bottom: 25px;
  }

  .left {
    padding: 30px 30px;
    font-size: 2rem;
    line-height: 8rem;
  }
  .built-with {
    padding: 50px 0 30px;
  }
  article {
    // background-color: darkseagreen;
    p {
      line-height: 3rem;
    }
  }
  .bb-style {
    background-color: #36603c;
    padding: 10px 6px;
    border: 1px solid white;
    line-height: 3.7rem;
  }

  .typing::after {
    content: "";
    position: absolute;
    padding-left: 1px;
    width: 2px;
    height: 79%;
    top: 1px;
    border-right: 2px solid white;
    animation: blink 0.4s infinite ease;
  }

  @keyframes blink {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }

  .tecs {
    padding: 10px 0;
  }

  .legend {
    padding: 10px 0;

    .wrapper {
      display: flex;
      justify-content: space-evenly;
      padding-top: 5px;
    }
  }

  .list {
    ul {
      text-align: start;
      padding: 10px 0 0 30px;
      line-height: 3rem;
    }
    li {
      list-style: disc;
    }
  }
  .ongoing {
    background-color: orange;
    padding: 0 5px;
    font-size: 2.2rem;
  }
}

.left {
  text-align: left;
}
