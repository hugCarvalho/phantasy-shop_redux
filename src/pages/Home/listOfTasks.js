export const listOfTasks = [
  {
    title: "Router: Setting React-router logic for page navigation",
    completed: "partial",
  },
  {
    title: "Async: fetching logic for fetching data from Breaking Bad API ",
    completed: true,
  },
  {
    title:
      "Async: Adapting received data to online shop concept by adding entries to it, amount, price, etc",
    completed: true,
  },
  {
    title: "Itemlist: Displaying data in item overview",
    completed: true,
  },
  {
    title: "Itemlist: Add `add to cart / remove from cart logic`",
    completed: true,
  },
  {
    title: "Itemlist: rate actor. Only if logged in",
    completed: false,
  },
  {
    title: "Itemlist: add to favorites. Must be logged in",
    completed: false,
  },
  {
    title: "Itemlist: pagination",
    completed: false,
  },
  {
    title: "Itemlist: filters",
    completed: false,
  },
  {
    title: "Itemlist: item discount - 2 prices: old & new",
    completed: false,
  },
  {
    title: "Itemlist: format currency",
    completed: false,
  },
  {
    title: "Cart: format currency",
    completed: false,
  },
  {
    title: "Cart: displaying number of products in cart on the fly ",
    completed: true,
  },
  {
    title: "Cart: add cart icon",
    completed: false,
  },
  {
    title: "Cart: display items",
    completed: true,
  },
  {
    title: "Cart: increase / decrease quantities ",
    completed: true,
  },
  {
    title: "Cart: calc item subtotal + calc cart total",
    completed: true,
  },
  {
    title: "Cart: add inventory logic",
    completed: false,
  },
  {
    title: "Cart: add tax + item discount + cart discount + enter voucher",
    completed: false,
  },
  {
    title:
      "Cart: add buy button and mockup behaviour(is login success else ask for login)",
    completed: false,
  },
  {
    title: "Register: do register form - html + css",
    completed: false,
  },
  {
    title: "Register: add logic - only frontend sim",
    completed: false,
  },
  {
    title: "Login: do login form - html + css ",
    completed: false,
  },
  {
    title: "Login: add logic to form - only frontend sim",
    completed: false,
  },
  {
    title: "Alerts: to be defined",
    completed: false,
  },
  {
    title: "Responsive: add media queries",
    completed: false,
  },
  {
    title: "CSS: make things prettier - Item List",
    completed: false,
  },
  {
    title: "CSS: make things prettier - Cart",
    completed: false,
  },
  {
    title: "CSS: make things prettier - Register",
    completed: false,
  },
  {
    title: "CSS: make things prettier - Login",
    completed: false,
  },
  {
    title: "CSS: make things prettier - Nav Bar",
    completed: false,
  },
  {
    title: "CSS: make things prettier - Home",
    completed: false,
  },
];
