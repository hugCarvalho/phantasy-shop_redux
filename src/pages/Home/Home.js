import React from "react";
import { listOfTasks } from "./listOfTasks";
import "./Home.scss";

const taskStatus = (status) => {
  if (status === true) return { color: "green" };
  else if (!status) return { color: "red" };
  else return { color: "orange" };
};

//Computation for report
const calcCategories = (list) => {
  const firstWordOfEachTitle = list.map((item) => item.title.split(" ")[0]);
  const categories = {};

  firstWordOfEachTitle.forEach((item) => {
    item = item.slice(0, item.length - 1); //removes colon
    if (!categories[item]) return (categories[item] = 1);
    else return (categories[item] += 1);
  });
  return Object.entries(categories);
};
const calcCompleted = (list) => list.filter((item) => item.completed).length;
const calcCompletedPercent = (list) =>
  Math.round((calcCompleted(list) * 100) / list.length);

function Home() {
  return (
    <div className="Home">
      <article>
        <h1>Important:</h1>
        <p>
          This is an{" "}
          <u>
            <span style={{ backgroundColor: "lightpink" }}>ongoing</span>
          </u>{" "}
          project whose focus is, primarily, behaviour with REACT (hooks) and
          REDUX.
        </p>
        <p>
          Graphical enhancements and responsive optimization will be added at a
          later stage, as well as optimized code.
        </p>
        <br></br>

        <p>
          <strong>Descr:</strong> This project consists on a{" "}
          <span style={{ backgroundColor: "lightpink" }}>
            phantasy online shop
          </span>{" "}
          where you can book days with actors of the Breaking Bad series (one of
          my favorites! ).{" "}
          <u>
            {" "}
            The data consists on a mixture of the Breaking Bad API data and
            object manipulation in order to add the necessary entries for a shop
            behaviour.
          </u>
        </p>
        <br></br>
        <p>
          Below, a list of the main libraries used, a dynamic list of the tasks
          for the project and also a simple dynamic report, just for fun...
        </p>
        <hr></hr>
      </article>
      <div className="tecs">
        <i>
          <strong>React-redux / React-router / React-hook-form</strong>
        </i>
      </div>
      <hr></hr>
      <div className="legend">
        <h5>Tasks</h5>
        <div className="wrapper">
          <span style={{ color: "green" }}>green: completed</span>
          <span style={{ color: "orange" }}>orange: partially completed</span>
          <span style={{ color: "red" }}>red: not began</span>
        </div>
      </div>
      <hr></hr>
      <section className="list">
        <h5>List</h5>
        <ul>
          {listOfTasks.map((item, i) => (
            <li key={i} style={taskStatus(item.completed)}>
              {item.title}
            </li>
          ))}
        </ul>
      </section>
      <section className="statistics">
        <h5>Report</h5>
        <p>{`There are a total of ${
          listOfTasks.length
        } tasks, from which ${calcCompleted(
          listOfTasks
        )} are completed (${calcCompletedPercent(listOfTasks)}%)  `}</p>
        <p>{`There are ${calcCategories(listOfTasks).length} categories:`}</p>
        {calcCategories(listOfTasks).map((item, i) => (
          <p key={i}>
            {item[0]} : {item[1]} {item[1] > 1 ? "tasks" : "task"}
          </p>
        ))}
      </section>
    </div>
  );
}

export default Home;
