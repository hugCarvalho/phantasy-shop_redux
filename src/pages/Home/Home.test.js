import React from "react";
import { shallow } from "enzyme";
import Home from "./Home";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";

Enzyme.configure({ adapter: new Adapter() });

test.skip("make snapshot component", () => {
  expect(shallow(<Home />)).toMatchSnapshot();
});

test("snapshot", () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});
