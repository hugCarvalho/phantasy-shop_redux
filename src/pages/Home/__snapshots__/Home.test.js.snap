// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`make snapshot component 1`] = `ShallowWrapper {}`;

exports[`snapshot 1`] = `
<div
  className="Home"
>
  <article>
    <h1>
      Important
    </h1>
    <p>
      This is an
       
      <u>
        <span
          className="ongoing"
        >
           !!! ongoing !!! 
        </span>
      </u>
       
      project whose focus is, primarily, 
      <u>
        behaviour
      </u>
       with REACT (hooks) and REDUX.
    </p>
    <p>
      Graphical enhancements and responsive optimization will be added at a later stage, as well as optimized code.
    </p>
    <br />
    <p>
      <strong>
        Descr:
      </strong>
       This project consists on a
       
      <span>
        phantasy online shop
      </span>
       where you can book days with actors of the Breaking Bad series (one of my favorites! ).
       
      <u>
         
        The data consists on a mixture of the Breaking Bad API data and object manipulation in order to add the necessary entries for a shop behaviour.
      </u>
    </p>
    <br />
    <hr />
  </article>
  <div
    className="tecs"
  >
    <i>
      <strong>
        React-redux / React-router / React-hook-form
      </strong>
    </i>
  </div>
  <hr />
  <div
    className="legend"
  >
    <h5>
      Tasks
    </h5>
    <div
      className="wrapper"
    >
      <span
        style={
          Object {
            "color": "green",
          }
        }
      >
        green: completed
      </span>
      <span
        style={
          Object {
            "color": "orange",
          }
        }
      >
        orange: partially completed
      </span>
      <span
        style={
          Object {
            "color": "red",
          }
        }
      >
        red: not began
      </span>
    </div>
  </div>
  <hr />
  <section
    className="list"
  >
    <h5>
      List
    </h5>
    <ul>
      <li
        style={
          Object {
            "color": "orange",
          }
        }
      >
        Router: Setting React-router logic for page navigation
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Async: fetching logic for fetching data from Breaking Bad API 
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Async: Adapting received data to online shop concept by adding entries to it, amount, price, etc
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Itemlist: Displaying data in item overview
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Itemlist: Add \`add to cart / remove from cart logic\`
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        Itemlist: rate actor. Only if logged in
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        Itemlist: add to favorites. Must be logged in
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        Itemlist: pagination
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        Itemlist: filters
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        Itemlist: item discount - 2 prices: old & new
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Itemlist: format currency
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Cart: displaying number of products in cart on the fly 
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Cart: add cart icon
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Cart: display items
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Cart: increase / decrease quantities 
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Cart: calc item subtotal + calc cart total
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Cart: add inventory logic
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        Cart: add tax + item discount + cart discount + enter voucher
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        Cart: add buy button and mockup behaviour(is login success else ask for login)
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Register: do register form - html + css
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        Register: add logic - only frontend sim
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Login: do login form - html + css 
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        Login: add logic to form - only frontend sim
      </li>
      <li
        style={
          Object {
            "color": "orange",
          }
        }
      >
        Alerts: to be defined
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        Responsive: add media queries
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        CSS: make things prettier - Item List
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        CSS: make things prettier - Cart
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        CSS: make things prettier - Register
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        CSS: make things prettier - Login
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        CSS: make things prettier - Nav Bar
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        CSS: make things prettier - Home
      </li>
    </ul>
  </section>
  <section
    className="statistics"
  >
    <h5>
      Report
    </h5>
    <p>
      There are a total of 31 tasks, from which 15 are completed (48%)  
    </p>
    <p>
      There are 9 categories:
    </p>
    <p>
      Router
       : 
      1
       
      task
    </p>
    <p>
      Async
       : 
      2
       
      tasks
    </p>
    <p>
      Itemlist
       : 
      8
       
      tasks
    </p>
    <p>
      Cart
       : 
      8
       
      tasks
    </p>
    <p>
      Register
       : 
      2
       
      tasks
    </p>
    <p>
      Login
       : 
      2
       
      tasks
    </p>
    <p>
      Alerts
       : 
      1
       
      task
    </p>
    <p>
      Responsive
       : 
      1
       
      task
    </p>
    <p>
      CSS
       : 
      6
       
      tasks
    </p>
  </section>
</div>
`;
